packages/settings-library/src/settings_library/base.py

```diff
@@ -1,74 +1,111 @@
-                field_obj.required = False
-            except ValidationError as e:
-                logger.error(
-                    (
-                        "Could not validate '%s', field '%s' "
-                        "contains errors, see below:\n%s"
-                        "\n======ENV_VARS=====\n%s"
-                        "\n==================="
-                    ),
-                    cls.__name__,
-                    default_cls.__name__,
-                    str(e),
-                    "\n".join(f"{k}={v}" for k, v in os.environ.items()),
+        @classmethod
+        def prepare_field(cls, field: ModelField) -> None:
+            super().prepare_field(field)
+
+            auto_default_from_env = field.field_info.extra.get(
+                "auto_default_from_env", False
+            )
+
+            field_type = field.type_
+            if args := get_args(field_type):
+                # TODO: skip all the way if none of these types
+                field_type = next(a for a in args if a != type(None))
+
+            if issubclass(field_type, BaseCustomSettings):
+
+                if auto_default_from_env:
+
+                    assert field.field_info.default is Undefined
+                    assert field.field_info.default_factory is None
+
+                    field.default_factory = create_settings_from_env(field)
+
+                    # TODO: doc why we are doing this?
+                    # Undefined required -> required=true
+                    # Undefined default and no factor -> default=None
+                    field.required = False
+
+            elif issubclass(field_type, BaseSettings):
+                raise ValueError(
+                    f"{cls}.{field.name} of type {field_type} must inherit from BaseCustomSettings"
                 )
-                raise e
 
-    @classmethod
-    def create_from_envs(cls):
-        """Constructs settings instance capturing envs (even for defaults) at this call moment"""
-
-        # captures envs here to build defaults for BaseCustomSettings sub-settings
-        default_fields = []
-        for name, field in cls.__fields__.items():
-            if issubclass(field.type_, BaseCustomSettings):
-                default_fields.append((name, field.type_))
-            elif issubclass(field.type_, BaseSettings):
+            elif auto_default_from_env:
                 raise ValueError(
-                    f"{name} field class {field.type_} must inherit from BaseCustomSettings"
+                    "auto_default_from_env=True can only be used in BaseCustomSettings subclasses"
+                    f"but field {cls}.{field.name} is {field_type} "
                 )
-        cls._set_defaults_with_default_constructors(default_fields)
 
-        # builds instance
-        obj = cls()
-        return obj
+    @classmethod
+    def create_from_envs(cls, **overrides):
+        # Legacy. More explicit and pylance does not get confused
```

Comment:
is this needed because this is how it is normally done? or we should still use it?

Reply:
it is legacy, so in principle no. 
Nonetheless, I leave you as an option since the code could be more informative.